docs/package_templates/cmake_package/all/conanfile.py

```diff
@@ -162,6 +165,18 @@ def package_info(self):
             self.cpp_info.system_libs.append("pthread")
             self.cpp_info.system_libs.append("dl")
 
+        # To export additional CMake variables, such as upper-case variables otherwise set by the project's *-config.cmake,
+        # you can copy or save a .cmake file under <prefix>/lib/cmake/ with content like
+        #     set(XYZ_VERSION ${${CMAKE_FIND_PACKAGE_NAME}_VERSION})
+        #     set(XYZ_INCLUDE_DIRS ${${CMAKE_FIND_PACKAGE_NAME}_INCLUDE_DIRS})
+        #     ...
+        # and set the following fields:
+        self.cpp_info.builddirs.append(os.path.join("lib", "cmake"))
+        cmake_module = os.path.join("lib", "cmake", "conan-official-variables.cmake")
+        self.cpp_info.set_property("cmake_build_modules", [cmake_module])
+        self.cpp_info.build_modules["cmake_find_package"] = [cmake_module]
+        self.cpp_info.build_modules["cmake_find_package_multi"] = [cmake_module]
```

Comment:
Matching the `*-config.cmake` output created by the project is often overlooked, imo, and it's difficult to find a good example on how to handle this as a newbie.
`${CMAKE_FIND_PACKAGE_NAME}` is preferable over a hard-coded package name as it allows the name to be overriden in CMakeDeps without breaking things.